bin/make_cfgs.py

```diff
@@ -0,0 +1,182 @@
+with records like this
+ 
+    ...
+    dhcp-host=aa:bb:cc:dd:68:7c,ses-client-3,1.2.3.13
+    ...
+    ptr-record=13.3.2.1.in-addr.arpa,ses-client-3.a.b.de
+    ...
+
+for conman generates file <target dir>/states/etc/conman.conf  based on 
+template  <target dir>states/etc/conman.conf.tmpl with records like this
+
+    CONSOLE name="ses-client-3" IPMIOPTS="U:<user>,P:<pass>" dev="ipmi:<ipmi server>"
+
+for salt generates roster file <target dir>/deploy.roster
+
+    ses-client-3.a.b.de:
+      host: 1.2.3.13
+      user: root
+
+for teuthology generates sql script <target dir>/sql/nodes.sql
+
+    INSERT INTO nodes (name, machine_type, mac_address, is_vm, locked, arch, description)
+        VALUES  ( 'ses-client-3.a.b.de' , 'client' ,
+            'aa:bb:cc:dd:67:16' , 'f' , 'f' , 'x86-64' , '' ) ;
+"""
+
+parser = argparse.ArgumentParser(description=description,
+        formatter_class=argparse.RawDescriptionHelpFormatter)
+
+
+def reorder_ip(ip):
+    ip_split = re.search('(\d+)\.(\d+)\.(\d+)\.(\d+)', ip)
+    print("ip_split:")
+    print(ip_split.groups())
+    return ip_split.groups()[3] + '.'\
+           + ip_split.groups()[2] + '.'\
+           + ip_split.groups()[1] + '.'\
+           + ip_split.groups()[0]
+
+# get host name from probably fqdn
+def hn(name):
+    return name.split('.')[0]
+
+
+# hardcoded but should be parametrized
+pxe_output_file = 'states/etc/dnsmasq/dnsmasq.d/network_nodes.conf'
+conman_tmpl = "states/etc/conman.conf.tmpl"
+conman_cfg = "states/etc/conman.conf"
+roster_file = "deploy.roster"
+
+bmc_user = ""
+bmc_pass = ""
+domain = ""
+
+sql_script_file = 'sql/nodes.sql'
+sql_line_prefix = 'INSERT INTO nodes ' \
+                  '(name, machine_type, mac_address, is_vm, ' \
+                  'locked, arch, description) VALUES '
+
+parser.add_argument('-c', dest='config_file', required=True,
+                    help='path to configuration file')
+
+parser.add_argument('-d', dest='target_dir', required=True,
+                    help='target directrory for generating files')
+
+args = parser.parse_args()
+
+network_manager.data_file = args.config_file
+nodes = network_manager.get_nodes()
+try:
+    os.mkdir(args.target_dir)
+except FileExistsError:
+    # it is ok if dir already exists
+    pass
+
+target_dir_prefix = args.target_dir + "/"
+
+bmc_user = network_manager.get_option('ipmi_user')
+bmc_pass = network_manager.get_option('ipmi_pass')
+domain = network_manager.get_option('domain')
+
+
+consoles = ""
+roster_recors = ""
+# ----------- iterating and preparing data ---------------
+pxe_node_lines = []
+pxe_node_ptr_lines = []
+nodes_sql_lines = []
+
+for n in nodes:
+    print('Process node' + n['node'])
+    
+    if n['ip_type'] == 'dynamic':
+        pxe_node_lines.append("dhcp-host={},{},{}"
+                          .format(n['mac'], hn(n['node']), n['ip']))
+        pxe_node_ptr_lines.append("ptr-record={}.in-addr.arpa,{}.{}"
+                              .format(reorder_ip(n['ip']),
```

Comment:
I thought it should be `reversed_ip` here.

Reply:
it was renamed previously